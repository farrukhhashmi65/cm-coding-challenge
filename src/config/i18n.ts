/* eslint-disable */

import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import english from '../localization/en.json'

i18n
  .use(initReactI18next)
  .init({
    resources: {
      en: {
        translation: {
          ...english
        }
      }
    },
    lng: 'en',
    fallbackLng: 'en',
    interpolation: {
      escapeValue: false
    }
  })

export default i18n
