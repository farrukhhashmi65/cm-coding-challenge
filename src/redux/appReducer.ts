import { combineReducers } from 'redux'
import absencesReducer from './reducer/absencesReducer'

// Combine Reducer - here we can add other reducers

export const appReducer = combineReducers({
  absencesReducer
})
