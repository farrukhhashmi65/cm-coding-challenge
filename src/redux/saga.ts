import { all } from 'redux-saga/effects'
import { watchAbsencesSaga } from './sagas/absencesSaga'

// Root Saga
export default function * (): any {
  yield all([
    watchAbsencesSaga()
  ])
}
