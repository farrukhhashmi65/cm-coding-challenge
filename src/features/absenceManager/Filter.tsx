import React, { useEffect } from 'react'
import Dropdown from '../../components/Dropdown'
import Grid from '@mui/material/Grid'
import Box from '@mui/material/Box'
import HighlightCard from '../../components/HighlightCard'
import Typography from '@mui/material/Typography'
import TextField from '@mui/material/TextField'
import Button from '@mui/material/Button'
import { useAppSelector, useAppDispatch } from '../../redux/reduxHooks'
import { useTranslation } from 'react-i18next'
import Container from '@mui/material/Container'
import { getAbsencesRequest, getAbsencesDataRequest } from '../../redux/actions/absencesActions'
import FilterIcon from '@mui/icons-material/FilterAltSharp'

interface Error {
  [key: string]: any
  startDate: string
  endDate: string
}

const Filter: React.FC<any> = (): JSX.Element => {
  const { t } = useTranslation()
  const [absenceType, setAbsenceType] = React.useState<string>('')
  const [startDate, setStartDate] = React.useState<string>('')
  const [endDate, setEndDate] = React.useState<string>('')
  const [active, setActive] = React.useState<boolean>(false)
  const [errors, setErrors] = React.useState<Error>({ startDate: '', endDate: '' })

  const {
    response
  } = useAppSelector((state: any) => state.absencesReducer.absencesData)

  const dispatch = useAppDispatch()

  const onSelectDropdown = (value: string): void => {
    setAbsenceType(value)
  }

  useEffect(() => {
    dispatch(getAbsencesRequest({}))
    dispatch(getAbsencesDataRequest())
  }, [])

  const onSubmitFilter = (): void => {
    if (startDate !== '' && endDate !== '' && startDate > endDate) {
      setErrors({ startDate: t('dateError'), endDate: t('dateError') })
      return
    }
    setErrors({ startDate: '', endDate: '' })
    setActive(true)
    dispatch(getAbsencesRequest({ startDate, endDate, absenceType }))
  }

  const onStartDateChange = (event: any): void => {
    setStartDate(event.target.value)
  }

  const onEndDateChange = (event: any): void => {
    setEndDate(event.target.value)
  }

  return (
        <>
            <Container fixed sx={{ mt: 4 }}>
                <Grid container spacing={2} >
                    <Grid item xl={3} lg={3} md={6} sm={12} xs={12}>
                        <Box sx={{ p: 1 }}>
                            <HighlightCard>
                                <Typography variant="subtitle2">
                                    {response?.totalAbsences ?? 0}
                                </Typography>
                                <Typography variant="caption" display="block">
                                    {t('totalAbsences')}
                                </Typography>
                            </HighlightCard>
                        </Box>
                    </Grid>
                    <Grid item xl={3} lg={3} md={6} sm={12} xs={12}>
                        <Box sx={{ p: 1 }}>
                            <Dropdown label={t('absenceType')}
                                dropdownList={response?.absencesTypes ?? []}
                                value={absenceType}
                                onSelect={onSelectDropdown}
                            />
                        </Box>
                    </Grid>
                    <Grid item xl={3} lg={3} md={6} sm={12} xs={12}>
                        <Box sx={{ p: 1 }}>
                            <TextField
                                id="date"
                                label={t('startDate')}
                                type="date"
                                fullWidth
                                InputLabelProps={{
                                  shrink: true
                                }}
                                onChange={onStartDateChange}
                                error={Boolean(errors?.startDate)}
                                helperText={errors?.startDate}
                            />
                        </Box>
                    </Grid>
                    <Grid item xl={3} lg={3} md={6} sm={12} xs={12}>
                        <Box sx={{ p: 1 }}>
                            <TextField
                                id="date"
                                label={t('endDate')}
                                type="date"
                                fullWidth
                                InputLabelProps={{
                                  shrink: true
                                }}
                                onChange={onEndDateChange}
                                error={Boolean(errors?.endDate)}
                                helperText={errors?.endDate}
                            />
                        </Box>
                    </Grid>
                    <Grid item xl={3} lg={3} md={6} sm={12} xs={12}>
                        <Box sx={{ p: 1 }}>
                            <Button variant="outlined" color={active ? 'secondary' : 'primary'} startIcon={<FilterIcon />} size="large" fullWidth sx={{ p: 1.7 }} onClick={() => onSubmitFilter()}>
                                {t('filter')}
                            </Button>
                        </Box>
                    </Grid>
                </Grid>
            </Container>
        </>
  )
}

export default Filter
