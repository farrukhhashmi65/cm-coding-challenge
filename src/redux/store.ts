import { configureStore } from '@reduxjs/toolkit'
import createSagaMiddleware from 'redux-saga'
import rootSaga from './saga'
import { appReducer } from './appReducer'
import logger from 'redux-logger'

const middleware = [logger]

// Adding the Redux Middleware

const sagaMiddleware = createSagaMiddleware()
middleware.push(sagaMiddleware)

export const store = configureStore({
  reducer: appReducer,
  middleware: [...middleware]
})

sagaMiddleware.run(rootSaga)

export type AppDispatch = typeof store.dispatch
export type RootState = ReturnType<typeof store.getState>
