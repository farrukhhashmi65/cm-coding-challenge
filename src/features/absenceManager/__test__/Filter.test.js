import * as React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import Filter from '../Filter'
import { Provider } from 'react-redux'
import { appReducer } from '../../../redux/appReducer'
import { I18nextProvider } from "react-i18next";
import i18n from '../../../config/i18n'

const store = configureStore({
    reducer: appReducer,
})


describe('Filter', () => {
    it('renders the filter form and calls the appropriate actions', () => {
        const { getByLabelText, getByText } = render(
            <Provider store={store}>
                <I18nextProvider i18n={i18n}>
                    <Filter />
                </I18nextProvider>
            </Provider>
        )
        const absenceTypeDropdown = getByLabelText('Absence Type')
        const startDateInput = getByLabelText('Start Date')
        const endDateInput = getByLabelText('End Date')
        const submitButton = getByText('Filter')

        fireEvent.click(absenceTypeDropdown)
        fireEvent.change(startDateInput, { target: { value: '2022-01-01' } })
        fireEvent.change(endDateInput, { target: { value: '2022-01-15' } })
        fireEvent.click(submitButton)

        expect(store.getState().absencesReducer.absencesData).toEqual({
            loading: true,
            error: null,
            response: null
          })
        
    })

    it('renders the filter snapshot', () => {
        const component = render(
            <Provider store={store}>
                <I18nextProvider i18n={i18n}>
                    <Filter />
                </I18nextProvider>
            </Provider>
        )
     
        expect(component).toMatchSnapshot();
    })
})