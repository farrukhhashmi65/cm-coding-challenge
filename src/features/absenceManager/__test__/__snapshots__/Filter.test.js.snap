// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders the filter snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-2dm2d1-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fnkheg-MuiPaper-root"
              >
                <h6
                  class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
                >
                  0
                </h6>
                <span
                  class="MuiTypography-root MuiTypography-caption css-11idbvn-MuiTypography-root"
                >
                  Total Absences
                </span>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <div
                  class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
                >
                  <label
                    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="false"
                    data-testid="dropdownLabel"
                    id="simple-select-label"
                  >
                    Absence Type
                  </label>
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    data-testid="selectInput"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="listbox"
                      aria-labelledby="simple-select-label simple-select"
                      class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                      id="simple-select"
                      role="button"
                      tabindex="0"
                    >
                      <span
                        class="notranslate"
                      >
                        â€‹
                      </span>
                    </div>
                    <input
                      aria-hidden="true"
                      class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                      tabindex="-1"
                      value=""
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                      data-testid="ArrowDropDownIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-yjsfm1"
                      >
                        <span>
                          Absence Type
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="date"
                  id="date-label"
                >
                  Start Date
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="date"
                    type="date"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Start Date
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="date"
                  id="date-label"
                >
                  End Date
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                    id="date"
                    type="date"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        End Date
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
          >
            <div
              class="MuiBox-root css-hpgf8j"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth css-ahv7fe-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FilterAltSharpIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 4c2.01 2.59 7 9 7 9v7h4v-7s4.98-6.41 7-9H3z"
                    />
                  </svg>
                </span>
                Filter
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-fixed css-2dm2d1-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-hpgf8j"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fnkheg-MuiPaper-root"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
              >
                0
              </h6>
              <span
                class="MuiTypography-root MuiTypography-caption css-11idbvn-MuiTypography-root"
              >
                Total Absences
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-hpgf8j"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  data-testid="dropdownLabel"
                  id="simple-select-label"
                >
                  Absence Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="selectInput"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="simple-select-label simple-select"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    id="simple-select"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      â€‹
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Absence Type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-hpgf8j"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="date"
                id="date-label"
              >
                Start Date
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="date"
                  type="date"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      Start Date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-hpgf8j"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="date"
                id="date-label"
              >
                End Date
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="date"
                  type="date"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      End Date
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-3 MuiGrid-grid-xl-3 css-72mr71-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-hpgf8j"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth css-ahv7fe-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeLarge css-htszrh-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FilterAltSharpIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 4c2.01 2.59 7 9 7 9v7h4v-7s4.98-6.41 7-9H3z"
                  />
                </svg>
              </span>
              Filter
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
